File: Data Structures & Algorithms/string-encode-and-decode/submission-10.js
class Solution {
    /**
     * @param {string[]} strs
     * @returns {string}
     */
    encode(strs) {
        if(!strs.length) return '0.0'
        return strs.join('0.0')
    }

    /**
     * @param {string} str
     * @returns {string[]}
     */
    decode(str) {
        if(str === '0.0') return []
        return str.split('0.0')
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-10.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ''

        prefix = strs[0]

        for s in strs:
            p = 0
            while p < min(len(prefix), len(s)) and prefix[p] == s[p]:
                p += 1

            prefix = s[:p] 

        return prefix